### PR TITLE
Feature fail http to https for known status code and substrings

### PR DESCRIPTION
**Pull request by mzpqnxow**
This implements what is described in #273. If zgrab2 http was invoked with `--retry-https`, the response code to a request is 400 *AND*  one of the three known substrings is present, consider it as a protocol failure to cause a retry as HTTPS

## How to Test
Use against an nginx or Apache server that speaks HTTPS. Specify `--retry-https` and the flag implemented here (`--fail-http-to-https`). Rather than storing the 400 response that suggests you retry the request as HTTPS, zgrab2 will automatically retry as HTTPS

## Notes & Caveats

This is not a generic solution- it's a specific workaround for specific HTTP servers- nginx and Apache. I believe that the substrings included have very good coverage for this sort of behavior based on some sample data I collected

I considered what impact this might have on performance, but I figured since the function this happens in is responsible for performing the hashing of the body the limited logic and string comparisons shouldn't be much overhead. To be on the safe side, I slice the string down to a smaller size since the sub-strings it looks for are always towards the beginning of the response

## Issue Tracking

Issue #273

### CODE REVIEW

File: modules/http/scanner.go [hunk 1]
```diff
@@ -446,6 +448,23 @@ func (scan *scan) Grab() *zgrab2.ScanError {
...
 
+	// Application-specific logic for retrying HTTP as HTTPS; if condition matches, return protocol error
+	if scan.scanner.config.FailHTTPToHTTPS && scan.results.Response.StatusCode == 400 && readLen < 1024 && readLen > 24 {
+		// Apache: "You're speaking plain HTTP to an SSL-enabled server port"
+		// NGINX: "The plain HTTP request was sent to HTTPS port"
+		var sliceLen int64 = 128
+		if readLen < sliceLen {
+			sliceLen = readLen
+		}
+		sliceBuf := bodyText[0:sliceLen]
```
Comment (vl4deee11) [inline modules/http/scanner.go:459]: you can replace with this `sliceBuf := bodyText[:sliceLen]`

Comment (mzpqnxow) [reply modules/http/scanner.go]: Thanks @vl4deee11, just committed. I'm very slowly picking up go as you can tell :>

**Comment by mzpqnxow** [comment]
@dadrian what are your thoughts on this? Is it a reasonable feature to include for the http module? I've gotten a good deal of use out of it and couldn't find any alternative aside from postprocessing results and running against and forcing https for the servers that behaved this way

**Comment by mzpqnxow** [comment]
poke :)

**Review by zakird — APPROVED** [review]

**Pull request merged**